scripts/dataproc-workflow-composer/clouddq_composer_dataplex_task_job.py

```diff
@@ -0,0 +1,226 @@
+                 ALL, \
+                 {CONFIGS_PATH}, \
+                --gcp_project_id={GCP_PROJECT_ID}, \
+                --gcp_region_id={GCP_BQ_REGION}, \
+                --gcp_bq_dataset_id={GCP_BQ_DATASET_ID}, \
+                --target_bigquery_summary_table={FULL_TARGET_TABLE_NAME}"
+        }
+    },
+    "trigger_spec": {
+        "type_": TRIGGER_SPEC_TYPE
+    },
+    "description": "Clouddq Airflow Task"
+}
+
+# for best practices
+YESTERDAY = datetime.datetime.now() - datetime.timedelta(days=1)
+
+# default arguments for the dag
+default_args = {
+    'owner': 'Clouddq Airflow task Example',
+    'depends_on_past': False,
+    'email': [''],
+    'email_on_failure': False,
+    'email_on_retry': False,
+    'retries': 1,
+    'retry_delay': datetime.timedelta(minutes=5),
+    'start_date': YESTERDAY,
+}
+
+
+def get_session_headers() -> tuple:
+    """
+    This method is to get the session and headers object for authenticating the api requests using credentials.
+    Args:
+    Returns: tuple
+    """
+    # getting the credentials and project details for gcp project
+    credentials, your_project_id = google.auth.default(scopes=["https://www.googleapis.com/auth/cloud-platform"])
+
+    # getting request object
+    auth_req = google.auth.transport.requests.Request()
+
+    print(credentials.valid)  # logger.debugs False
+    credentials.refresh(auth_req)  # refresh token
+    # cehck for valid credentials
+    print(credentials.valid)  # logger.debugs True
+    auth_token = credentials.token
+    print(auth_token)
+
+    headers = {
+        'Accept': 'application/json',
+        'Content-Type': 'application/json',
+        'Authorization': 'Bearer ' + auth_token
+    }
+
+    with requests.Session() as session:
+        session.auth = OAuth2BearerToken(auth_token)
+
+    return (session, headers)
+
+
+def get_clouddq_task_status() -> str:
+    """
+    This method will return the job status for the task.
+    Args:
+    Returns: str
+    """
+    session, headers = get_session_headers()
+    res = session.get(
+        f"{DATAPLEX_ENDPOINT}/v1/projects/{DATAPLEX_PROJECT_ID}/locations/{DATAPLEX_REGION}/lakes/{DATAPLEX_LAKE_ID}/tasks/{DATAPLEX_TASK_ID}/jobs",
+        headers=headers)
+    print(res.status_code)
+    print(res.text)
+    resp_obj = json.loads(res.text)
+    if res.status_code == 200:
+
+        if (
+                "jobs" in resp_obj
+                and len(resp_obj["jobs"]) > 0
+                and "state" in resp_obj["jobs"][0]
+        ):
+            task_status = resp_obj["jobs"][0]["state"]
+            return task_status
+    else:
+        return "FAILED"
+
+
+def _get_dataplex_job_state() -> str:
+    """
+    This method will try to get the status of the job till it is in either 'SUCCEEDED' or 'FAILED' state.
+    Args:
+    Returns: str
+    """
+    task_status = get_clouddq_task_status()
+    while (task_status != 'SUCCEEDED' and task_status != 'FAILED'):
```

Comment:
This should check for all terminal states. For example, if the status is ABORTED, then this will be infinite loop.